### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by IvanitskyiVolodymyr**

### CODE REVIEW

File: Backend/Application/ApplicationDependencyInjection.cs [hunk 1]
```diff
@@ -26,8 +38,8 @@ public static void ConfigureMapping(this IServiceCollection services)
 
         public static void ConfigureJwt(this IServiceCollection services, IConfiguration configuration)
         {
-            var jwtConfig = configuration.GetSection("JwtConfig");
-            var secretKey = jwtConfig["secret"];
+            var jwtConfig = configuration.GetSection(nameof(JwtConfig));
```
Comment (byme8) [inline Backend/Application/ApplicationDependencyInjection.cs:41]: You can use `` configuration.GetSection(nameof(JwtConfig)).Get<JwtConfig>();`` to get deserialized instance.

File: Backend/Application/Services/ScheduleService.cs [hunk 2]
```diff
@@ -14,50 +15,58 @@ public class ScheduleService : IScheduleService
...
-                             SubjectName = left.SubjectName,
-                             Place = left.Place,
-                             Status = sub?.Status   
-                         }).ToList();
+            var dict = new Dictionary<int, Attendance>();
+
+            foreach(var studAttend in studentAttendances)
+            {
+                dict.Add(studAttend.ScheduleId, studAttend);
+            }
```
Comment (byme8) [inline Backend/Application/Services/ScheduleService.cs:56]: What about:
``` csharp
var dict = studentAttendances.ToDictionary(o => o.ScheduleId);
```
Also name `` dict `` is not very informative. I guess studentAttendancesById would be better.

File: Backend/Application/Services/ScheduleService.cs [hunk 3]
```diff
@@ -14,50 +15,58 @@ public class ScheduleService : IScheduleService
...
+
+            var scheduleAttendances = _mapper.Map<List<ScheduleAttendanceDto>>(studentSchedule);
+
+            foreach(var s in scheduleAttendances)
+            {
+                Attendance? attendance = null;
+                s.Status = dict.TryGetValue(s.ScheduleId, out attendance) == true ? attendance?.Status : null;
+            }
+
+            return scheduleAttendances;
```
Comment (byme8) [inline Backend/Application/Services/ScheduleService.cs:66]:
``` csharp
s.Status = dict.GetValueOrDefault(s.ScheduleId)?.Status;
```

File: Backend/Application/Services/ScheduleService.cs [hunk 4]
```diff
@@ -89,14 +98,22 @@ public async Task<IEnumerable<StudentAttendanceDto>> GetAttendancesWithStudent(i
...
+                dict.Add(a.StudentId, a);
+            }
+
+            var studentAttendances = _mapper.Map<List<StudentAttendanceDto>>(students);
+
+            foreach (var s in studentAttendances)
+            {
+                Attendance? attendance = null;
+                 s.Attendance = dict.TryGetValue(s.Student.StudentId, out attendance) ? attendance : null;
+            }
```
Comment (byme8) [inline Backend/Application/Services/ScheduleService.cs:114]: The same issues like with scheduleAttendances

File: Backend/Application/ApplicationDependencyInjection.cs (hunk 1 above)
Comment (IvanitskyiVolodymyr) [reply Backend/Application/ApplicationDependencyInjection.cs:41]: Thanks, I changed it

File: Backend/Application/Services/ScheduleService.cs (hunk 2 above)
Comment (IvanitskyiVolodymyr) [reply Backend/Application/Services/ScheduleService.cs]: Cool method `ToDictionary()`. I will definitely use it in the future

**Comment by byme8** [comment]
Looks good

**Review by byme8 — APPROVED** [review]
Looks good

**Pull request merged**